### PR TITLE
fix(api): check if an application is available on the root node of a workflow as code

### PR DESCRIPTION
**Pull request by fsamin**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: sdk/error.go [hunk 1]
```diff
@@ -348,6 +349,7 @@ var errorsAmericanEnglish = map[int]string{
 	ErrInvalidJobRequirementWorkerModelCapabilitites.ID: "Invalid job requirements: the worker model does't match with the binary requirements",
 	ErrMalformattedStep.ID:                              "Malformatted step",
 	ErrVCSUsedByApplication.ID:                          "Repository manager still used by an application",
+	ErrApplicationMandatoryOnWorkflowAsCode.ID:          "An application repository is mandatory on the workflow root",
```
Comment (sguiheux) [inline sdk/error.go]:
```suggestion
	ErrApplicationMandatoryOnWorkflowAsCode.ID:          "An application linked to a git repository is mandatory on the workflow root",
```

File: sdk/error.go [hunk 2]
```diff
@@ -513,6 +515,7 @@ var errorsFrench = map[int]string{
 	ErrInvalidJobRequirementWorkerModelCapabilitites.ID: "Pré-requis de job invalide: Le modèle de worker ne dispose pas de binaires suffisants",
 	ErrMalformattedStep.ID:                              "Étape malformée",
 	ErrVCSUsedByApplication.ID:                          "Le gestionnaire de dépot est encore utilisé par une application",
+	ErrApplicationMandatoryOnWorkflowAsCode.ID:          "Un dépôt d'application est obligatoire à la racine du workflow",
```
Comment (sguiheux) [inline sdk/error.go]:
```suggestion
	ErrApplicationMandatoryOnWorkflowAsCode.ID:          "Une application liée à un dépôt GIT est obligatoire à la racine du workflow",
```

**Comment by ovh-cds** [comment]
CDS Report build-ui#9295.0 ✘  
* Compile
  * Build UI ✔ 
  * TU All -   
  * TU model ✔ 
  * TU service ✔ 
  * TU shared ✔ 
  * TU store ✔ 
  * TU views ✘

**Comment by fsamin** [comment]
close #4214

**Pull request merged**